blosc/blosc2.c

```diff
@@ -1963,12 +1966,28 @@ static int initialize_context_decompression(blosc2_context* context, blosc_heade
     return BLOSC2_ERROR_DATA;
   }
 
-  context->runlen_type = (header->blosc2_flags >> 4) & BLOSC2_RUNLEN_MASK;
+  context->special_type = (header->blosc2_flags >> 4) & BLOSC2_SPECIAL_MASK;
+  if (context->special_type > BLOSC2_SPECIAL_LASTID) {
+    BLOSC_TRACE_ERROR("Unknown special values ID (%d) ",
+                      context->special_type);
+    return BLOSC2_ERROR_DATA;
+  }
+
+  int memcpy = (context->header_flags & (uint8_t) BLOSC_MEMCPYED);
```

Comment:
Should not use function names as variable names.

Reply:
Fixed in 3dedb12